arangodb-net-standard/CollectionApi/CollectionApiClient.cs

```diff
@@ -267,5 +269,200 @@ public virtual async Task<GetCollectionFiguresResponse> GetCollectionFiguresAsyn
+            if (body == null || body.Count < 1)
+            {
+                throw new ArgumentException("body is required", "body");
+            }
+            string uriString = _collectionApiPath + "/" + WebUtility.UrlEncode(collectionName) + "/responsibleShard";
+            var content = GetContent(body, new ApiClientSerializationOptions(true, true));
+            using (var response = await _transport.PutAsync(uriString, content).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<PutDocumentShardResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Returns the shard ids of a collection.
+        /// This method is only available in a cluster Coordinator.
+        /// GET /_api/collection/{collection-name}/shards
+        /// </summary>
+        /// <param name="collectionName">Name of the collection.</param>
+        /// <returns></returns>
+        public virtual async Task<GetCollectionShardsResponse> GetCollectionShardsAsync(string collectionName)
+        {
+            if (string.IsNullOrWhiteSpace(collectionName))
+            {
+                throw new ArgumentException("collectionName is required", "collectionName");
+            }
+            string uriString = _collectionApiPath + "/" + WebUtility.UrlEncode(collectionName) + "/shards";
+            using (var response = await _transport.GetAsync(uriString).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetCollectionShardsResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Returns the shard ids of a collection.
+        /// This method is only available in a cluster Coordinator.
+        /// The response also contains shard IDs as object attribute 
+        /// keys, and the responsible servers for each shard mapped 
+        /// to them. The leader shards will be first in the arrays.
+        /// GET /_api/collection/{collection-name}/shards?details=true
+        /// </summary>
+        /// <param name="collectionName">Name of the collection.</param>
+        /// <returns></returns>
+        public virtual async Task<GetCollectionShardsDetailedResponse> GetCollectionShardsWithDetailsAsync(string collectionName)
+        {
+            if (string.IsNullOrWhiteSpace(collectionName))
+            {
+                throw new ArgumentException("collectionName is required", "collectionName");
```

Comment:
A small tip: we can use `nameof(collectionName)` to keep the variable name in sync with the exception parameter. It doesn't warrant changing it but just thought I would point it out.

Reply:
Good point. I've fixed it. :-)